### PR TITLE
support keep logs

### PR DESCRIPTION
**Pull request by vhvb1989**
Add support for `progress log` component to stop and keep the logs in the terminal

### CODE REVIEW

File: cli/azd/pkg/input/console.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ type Console interface {
 	// Use the returned io.Writer to produce the output within the previewer
 	ShowPreviewer(ctx context.Context, options *ShowPreviewerOptions) io.Writer
 	// Finalize the preview mode from console.
-	StopPreviewer(ctx context.Context)
+	StopPreviewer(ctx context.Context, keepLogs bool)
```
Comment (weikanglim) [inline cli/azd/pkg/input/console.go:77]: Any thoughts about keeping the API simple by having `keepLogs` be simply part of `ShowPreviewerOptions` when the previewer is constructed?

The only downside is that you can't decide later on whether to keep logs or not. Maybe that is indeed the requirement here.

File: cli/azd/pkg/input/progress_log.go [hunk 2]
```diff
@@ -118,27 +118,29 @@ func (p *progressLog) Start() {
 
 // Stop clears the screen from any previous output and clear the buffer.
```
Comment (weikanglim) [inline cli/azd/pkg/input/progress_log.go:119]: I think there's several places in this file where comments are made outdated by the change here.

File: cli/azd/pkg/input/progress_log.go [hunk 3]
```diff
@@ -118,27 +118,29 @@ func (p *progressLog) Start() {
...
-	tm.Flush()
+		// header
+		if p.header != "" {
+			tm.MoveCursorUp(2)
+			clearLine()
+		}
+
+		tm.Flush()
+	}
 	p.output = nil
```
Comment (weikanglim) [inline cli/azd/pkg/input/progress_log.go:144]: A really naive question here... Do we at least need to do something along the lines of `Flush` here to make output show up before clearing it?

**Review by weikanglim — APPROVED** [review]
LGTM. Some questions but nothing blocking

File: cli/azd/pkg/input/console.go (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/pkg/input/console.go:77]: The intention is to decide keeping or not the logs at the end. 

I considered adding a new bag of options for stop, like `StopPreviewerOptions`, but I will wait if there's ever some other setting that only applies to the stopping process

File: cli/azd/pkg/input/progress_log.go (hunk 3 above)
Comment (vhvb1989) [reply cli/azd/pkg/input/progress_log.go:144]: The flush() call happens for every Write (at the end), to refresh the output after updating the buffer.
When the component is stopped, we only need to call flush() if we make changes to the current buffer and need to re-paint the screen. For the case of keeping the logs, we want to skip this step

**Pull request merged**